### PR TITLE
Feature/claim command

### LINKED ISSUE #2341: Can't place building within "workingRangeTownHallChunks" range after upgrade
### Minecolonies version

version 1.12.2-0.8.7913, SSP

### Expected behavior

Changing "workingRangeTownHallChunks" in minecolonies.cfg should change the distance from colony center that a new building can be placed.

### Actual behaviour

A new building cannot be placed beyond 8 chunks of colony center (default setting,) regardless of value in "workingRangeTownHallChunks."  

### Steps to reproduce the problem

1. Upgrade an existing colony (mine started in version 1.12.2.-0.8.5081 and has been upgraded frequently) to a version of minecolonies after the "workingRangeTownHall" setting was changed to "workingRangeTownHallChunks."  Behavior is not observed in new worlds - those seem to work fine.

2. Change workingRangeTownHallChunks to a different number of chunks, restart Minecraft.

3. Actual build distance remains 8 chunks from colony center (per coordinates found in "center" in colony1.dat,) regardless of setting.  Other configuration settings seem to behave normally (although not exhaustively tested.)

FYI - entire colony is chunk loaded.  Diamond border of colony displays at the expected range in "workingRangeTownHall" when build tool used, and existing buildings built before upgrade continue to work - only no new buildings can be placed.

*****
A big THANKS to the devs for continued work on this Mod.  I can't remember the last time a mod so dramatically (and positively) changed the way I play the game.  Well done!  Although I now have 83 colonists that are very concerned about the unemployment rate due to a lack of build-able land.  Every day, idle youth gather in the town hall and are observed drinking, smoking, discussing the "good life" of the barbarians and planning their next "creeper tipping" outing to the mines.  This will only lead to trouble...  ;)

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #2341

Creates a command to manually claim an area as an admin (in a certain range).

# Changes proposed in this pull request:
- Creates claim command.
- Creates command constants which make sense.
- Move a bunch of constants over to command constants.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java [hunk 1]
```diff
@@ -270,7 +270,7 @@ public Object parse(@NotNull final MinecraftServer server, @NotNull final IComma
             case CITIZEN:
                 return parseCitizenDataValue(parsedHolders, potentialArgumentValue);
             case STRING:
-                return potentialArgumentValue;
+                return potentialArgumentValue.isEmpty() ? null : potentialArgumentValue;
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java:274]: Either mark this method as Nullable or revert this Change.

**Review by marchermans — CHANGES_REQUESTED** [review]
Check the method declaration for that Change there,

File: src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java (hunk 1 above)
Comment (mkienenb) [reply src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java:274]: Yes, that's my fault.   This method is Nullable.  All of the parse*() methods this calls are also Nullable and should be marked as such.

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java:274]: I added all of them in my last commit.
I also added a test case.

Btw @mkienenb great job on the command system, it is really easy to use and add to it.

File: src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java [hunk 2]
```diff
@@ -113,7 +113,7 @@ public static boolean isPlayerOpped(@NotNull final ICommandSender sender)
      * @param theCommand which command to check if the player can use it
      * @return boolean
      */
-    public boolean canCommandSenderUseCommand(final Commands theCommand)
+    public static boolean canCommandSenderUseCommand(final Commands theCommand)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java:116]: MAJOR Reduce the number of returns of this method 14, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java [hunk 3]
```diff
@@ -283,6 +284,7 @@ public Object parse(@NotNull final MinecraftServer server, @NotNull final IComma
      * We are returning null to indicate that no boolean value could be parsed.
      */
     @SuppressWarnings({"squid:S2447"})
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java:286]: INFO Suppressing the 'squid:S2447' warning is not allowed [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/ClaimChunksCommand.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
+package com.minecolonies.coremod.commands.colonycommands;
+
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.commands.ActionMenu;
+import com.minecolonies.coremod.commands.IActionCommand;
+import com.minecolonies.coremod.util.TeleportToColony;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/ClaimChunksCommand.java:7]: MINOR Remove this unused import 'com.minecolonies.coremod.util.TeleportToColony'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 80 issues
* ![CRITICAL][CRITICAL] 4 critical
* ![MAJOR][MAJOR] 61 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 8 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L977](https://github.com/Minecolonies/minecolonies/blob/334d96b29f5c24988b8ed60a9cd5d2fc57cf8d79/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L977): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [ActionMenuType.java#L40](https://github.com/Minecolonies/minecolonies/blob/334d96b29f5c24988b8ed60a9cd5d2fc57cf8d79/src/main/java/com/minecolonies/coremod/commands/ActionMenuType.java#L40): Define a constant instead of duplicating this literal "player" 6 times. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ActionMenuType.java#L106](https://github.com/Minecolonies/minecolonies/blob/334d96b29f5c24988b8ed60a9cd5d2fc57cf8d79/src/main/java/com/minecolonies/coremod/commands/ActionMenuType.java#L106): Define a constant instead of duplicating this literal "colony" 16 times. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [ActionMenuType.java#L115](https://github.com/Minecolonies/minecolonies/blob/334d96b29f5c24988b8ed60a9cd5d2fc57cf8d79/src/main/java/com/minecolonies/coremod/commands/ActionMenuType.java#L115): Define a constant instead of duplicating this literal "citizen" 3 times. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/334d96b29f5c24988b8ed60a9cd5d2fc57cf8d79/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/334d96b29f5c24988b8ed60a9cd5d2fc57cf8d79/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/334d96b29f5c24988b8ed60a9cd5d2fc57cf8d79/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/334d96b29f5c24988b8ed60a9cd5d2fc57cf8d79/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/334d96b29f5c24988b8ed60a9cd5d2fc57cf8d79/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/334d96b29f5c24988b8ed60a9cd5d2fc57cf8d79/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): class "ColonyManager" has 58 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java [hunk 5]
```diff
@@ -283,6 +284,7 @@ public Object parse(@NotNull final MinecraftServer server, @NotNull final IComma
...
+    @Nullable
     private static Boolean parseBoolean(final String potentialArgumentValue)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java:288]: MAJOR Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

**Review by marchermans — APPROVED** [review]
